apps/viewer/src/app.tsx

```diff
@@ -98,6 +89,20 @@ const App = () => {
     }
   }, [mainSource?.streamQualityOptions.length]);
 
+  const changeMainSource = async (sourceId: string) => {
```

Comment:
we also need to unproject the source's remote track that is going to be projected to main stream. I see we still just hide that track without unprojecting.